models.py

```diff
@@ -149,6 +149,26 @@ def deserialize(self, data, initialize_property=True):
                 self.restock_amt = DEFALUT_RESTOCK_AMT
 
         return self
+
+    def deserialize_restock(self, data, initialize_property=True):
```

Comment:
Hi, it seems the 'initialize_property' is never used, so maybe delete it?

Also this function seems to "deserialize the JSON string passed in by restock action" + "do the restocking", the 1st part seems to be more of a server side job than a model side job. Maybe we could separate these 2 parts and only let the server handle the deserialization and let the model do the restocking? 

The reason I'm not having trouble with function "deserialize" or "deserialize_update" is because that function deals with JSON string which is supposed to be the serialization form of the model ProductInformation, thus it seems redundant for the server side to do the input checking. But this case (restock) seems to be a different situation.

Still this is just bit of my feeling, I'm ok if you want to keep it this way since otherwise it might involve major change in this PR.